# MCMC-CE algorithm for the estimation of small p-values with applications in genomic data analysis
This repository contains R programs that implement the MCMC-CE algorithm for the estimation of small p-values and generate the results in our paper describing the algorithm titled *Accurate and Efficient Calculation of Small P-Values with the Cross-Entropy Method: Applications in Genomic Data Analysis* by Shi *et al* (The arXiv preprinted version of the manuscript is available at: https://arxiv.org/abs/1803.03373). Supplementary tables with the manuscript are also provided here. As the paper has not formally published yet, all the programs and results should only be used for academic purpose and should NOT be distributed without the permission of the authors (contact: yshi@augusta.edu).

The *R_programs.zip* file contains all the programs. After unzip the file, there are 6 folders:
1. The *simulation1_quadratic_function_of_MVN* folder contains R programs generating the simulation results with the quadratic form statistics based on the chi-squared distribution as reported in Section 3.1 of the Main Text and Section 2.1 of the Supplementary Text of the manuscript.
2. The *simulation2_linear_function_of_MVN* folder contains R programs generating the simulation results with the ratio statistic based on the standard Cauchy distribution as reported in Section 3.1 of the Main Text and Section 2.1 of the Supplementary Text of the manuscript.
3. The *simulation3_different_NandM* folder contains R programs generating the simulation results for the sensitivity analysis with different numbers of random samples in the parameter updating step and the estimating step (*N* and *M*). The relavant texts and results are in Section 3.1 of the Main Text, Section 2.2 of the Supplementary Text and Supplementary Table S6 of the manuscript.
4. The *Example1* folder contains R programs for generating the results of "Example 1: Gene set/pathway enrichment analysis" as reported in Section 3.2 of the Main Text of the manuscript. The RNA-Seq gene expression data and the clinical data for the melanoma patients are obtained from the TCGA data portal: https://portal.gdc.cancer.gov/ . Note that we used the data from the "Legacy Archive".
5. The *Example2* folder contains R programs for generating the results of "Example 2: GWAS â€“ joint testing a group of genetic markers in a genomic region" as reported in Section 3.2 of the Main Text of the manuscript. The mouse GWAS data are available from: https://wp.cs.ucl.ac.uk/outbredmice/heterogeneous-stock-mice/
6. The *Example3* folder contains R programs for generating the results of "Example 3: Ratio statistic in differential gene expression analysis" as reported in Section 3.2 of the Main Text of the manuscript. The RNA-Seq data of the salivary adenoid cystic carcinomai patients are available from the NCBI Sequence Read Archive using accession number SRP059557.

The programs were run and tested with R version 3.5.1 on Intel Xeon X5550 2.67GHz CPU with Scientific Linux operating system.

The *Supplementary_Tables.zip* file contains the Supplementary Tables S1-S9 of the manuscript.
